simulator/func_sim/func_sim.cpp

```diff
@@ -28,6 +28,10 @@ typename FuncSim<ISA>::FuncInstr FuncSim<ISA>::step()
     // fetch instruction
     FuncInstr instr = mem->fetch_instr( PC);
 
+    // set sequence_id
+    instr.set_sequence_id(sequence_id);
```

Comment:
To significantly simplify debugging, you may append disassembly to id inside this function, at least for MIPS instruction.
Use ostringstream to append COW string, there are plenty of examples in `mips_instr.cpp`
I would prefer following formatting for dumping: " {3423} ".